gems/smithy/lib/smithy.rb

```diff
@@ -19,10 +21,10 @@ module Smithy
 
   class << self
     # Generates a Ruby artifact from a Smithy model.
-    # @param plan [Smithy::Plan] the plan to generate the artifact from
+    # @param [Plan] plan The plan to generate the artifact from.
     def smith(plan)
       artifact = Smithy::Forge.forge(plan)
-      Smithy::Polish.descendants.each { |p| p.new(plan).polish(artifact) }
+      plan.polishes.each { |polish| polish.polish(artifact) }
```

Comment:
I'm struggling a little bit with why we have separate Welds and Polishes.  Why are these separate extension points - couldn't "polish" just be a hook on  a weld?

Reply:
Polish required instances because it uses Thor actions. Weld now has instances but doesn't include thor. I'm not sure if we should keep the distinction anymore or not.